recipes/cppwinrt/all/conanfile.py

```diff
@@ -0,0 +1,107 @@
+import os
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get
+from conan.tools.microsoft import MSBuild, MSBuildDeps, MSBuildToolchain, VCVars, check_min_vs, is_msvc, vs_layout
+
+required_conan_version = ">=1.53.0"
+
+
+class CppWinRtConan(ConanFile):
+    name = "cppwinrt"
+    description = "C++17 language projection for Windows Runtime"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/microsoft/cppwinrt"
+    topics = ("native", "cpp", "winrt")
+    settings = "os", "arch", "compiler", "build_type"
+
+    def layout(self):
+        vs_layout(self)
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def requirements(self):
+        self.requires("winmd/1.0.210629.2")
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    def validate(self):
+        if self.info.settings.compiler.cppstd:
```

Comment:
The advice around `self.info` changed recently is should only be used in `package_id` as of now

```suggestion
        if self.settings.get_safe("compiler.cppstd"):
```